src/js/components/menu-bar.react.js

```diff
@@ -29,11 +29,15 @@ const _clickOpenFile = function () {
 };
 
 const _clickOpenGist = function () {
-    openGistModal.show();
+    EditorIO.checkSaveStateThen(() => {
+        ReactDOM.render(<OpenGistModal />, document.getElementById('modal-container'));
```

Comment:
Here too... makes sense now how it all starts linking up. Pretty cool. And we know they get unmounted when a new component mounts on the same div.